[blog]: Expose OTEL Collector with Gateway API and mTLS

vipinvkmenon opened the pull request:
This PR introduces a blog that provides a step by step guide to securely expose OTEL collector in Kubernetes externally using the new Kubernetes Gateway API and mTLS

- Content that covers intro to gateways, mTLS, scenario 
- Setup Scenario
- Image for the scenario
- Step-by-step walkthrough with configuration
- Refcache (new links)
- Closes #6502 

---

**Preview**: https://deploy-preview-6655--opentelemetry.netlify.app/blog/2025/expose-otel-collector-gateway-api/

theletterf commented:
@vipinvkmenon Thanks! Could you please address the cSpell errors? https://opentelemetry.io/docs/contributing/style-guide/#spell-checking

svrnm commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,445 @@
+date: 2025-04-6
+author: >
+  [Vipin Vijaykumar](https://github.com/vipinvkmenon)
+sig: OpenTelemetry JS
```
```suggestion
sig: OpenTelemetry JS
```

Based on #6502, this is not the sponsoring SIG, either @open-telemetry/sig-end-user-approvers, but I also think there has to be support from @open-telemetry/operator-approvers and @open-telemetry/collector-approvers

vipinvkmenon replied:
Have updated to End User SIG

vipinvkmenon replied:
Anything to add in the sig?

svrnm commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,445 @@
+cSpell:ignore: Danielson
+---
+
+# Securely Exposing Your  OTEL Collector in Kubernetes with Gateway API and mTLS
```
You don't need to add your h1 title once again, this will be added based on the `title` above.

svrnm commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,445 @@
+---
+title: Securely Exposing Your  OTEL Collector in Kubernetes with Gateway API and mTLS
```
```suggestion
title: Securely Exposing Your  OTEL Collector in Kubernetes with Gateway API and mTLS
linkTitle: tbd
```

Please come up with a short title, that will be used for the link and in the navigation. Most titles do only have a few words

svrnm commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,445 @@
+---
+
+# Securely Exposing Your  OTEL Collector in Kubernetes with Gateway API and mTLS
+**Goal:** 
+
+Exposing an OpenTelemetry (OTEL) Collector running inside Kubernetes to the outside world securely, using the Kubernetes Gateway API and mutual TLS (mTLS) for authentication and encryption. 
```
```suggestion
The goal of this blog post is to demonstrate, how you can expose an OpenTelemetry (OTel) Collector running inside Kubernetes to the outside world securely, using the Kubernetes Gateway API and mutual TLS (mTLS) for authentication and encryption. 
```

Also add links to the Kubernetes Gateway API

svrnm commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,445 @@
+
+As observability becomes increasingly critical in modern distributed systems, centralizing telemetry data via an OTEL Collector is common practice. Often, services or agents running *outside* your Kubernetes cluster need to send data *to* this collector. Exposing internal services requires careful consideration of security and standardization. This is where the Kubernetes Gateway API and mTLS shi ...
+
+Typicall this kind of a setup is useful when you have applications/ workloads that are external to the cluster and you need to collect the telemetry data. Some examples:
```
```suggestion
Typically this kind of a setup is useful when you have applications/ workloads that are external to the cluster and you need to collect the telemetry data. Some examples:
```

Our spell checker provides you with some suggestions on wording issues, please review them and fix them accordingly (e.g. also the repeated issue of "OTEL" vs "OTel"

svrnm commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,445 @@
+Typicall this kind of a setup is useful when you have applications/ workloads that are external to the cluster and you need to collect the telemetry data. Some examples:
+
+- **Hybrid Cloud/On-Premise Environments:** Applications or servers running in a traditional data center, or a different cloud or external to your Kubernetes cluster need to forward their metrics, traces, or logs to your central observability solution.
+-  **Multi-Cluster Telemetry Aggregation:** In a setup with multiple Kubernetes clusters, you might designate one cluster to host the primary OTEL collector. Collectors in other "spoke" clusters would act as clients, exporting data to this central collector via its external endpoint.
```
```suggestion
- **Multi-Cluster Telemetry Aggregation:** In a setup with multiple Kubernetes clusters, you might designate one cluster to host the primary OTEL collector. Collectors in other "spoke" clusters would act as clients, exporting data to this central collector via its external endpoint.
```

svrnm reviewed (changes requested):
Providing a first pass review.

It would be great to have a review of this content from subject matter experts (@open-telemetry/collector-approvers and @open-telemetry/operator-approvers)

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,445 @@
+2.  **`kubectl`:** Configured to interact with your cluster.
+3.  **`helm`:**  Configured to install helm charts.
+4.  **A Gateway API Implementation:** We'll use **Istio** in this example. Other implementations like Contour, Nginx Gateway Fabric,   etc., would also work with potentially minor configuration adjustments.
+5.  **`openssl`:** For generating certificates.
```
Done

vipinvkmenon commented:
Hi @theletterf @svrnm @danielgblanco

Your support and feedback are highly appreciated.
I have incorporated the same and made the necessary corrections.

But I'm a bit confused about this REFCACHE PR check.... 
on running `npm run test-and-fix` and then checking the `git diff`, I see the following
However, these don't correlate to files I've changed or related to the PR.
```
git status
On branch issue-6502
Your branch is up to date with 'origin/issue-6502'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   content/es/docs/contributing/style-guide.md
	modified:   content/ja/docs/contributing/announcements.md
	modified:   content/ja/docs/contributing/pull-requests.md
	modified:   content/ja/docs/contributing/style-guide.md
	modified:   content/zh/docs/contributing/localization.md
	modified:   static/refcache.json

```

Do I add these?

Also, I've changed the SIG to  `End-User`.  
I've tried shortening the title and list title. (open to changing it or shorter titles suggestions are welcome :) )

Looking forward to your feedbacks

svrnm commented:
> Do I add these?

Please remove the changes to those files. I am not exactly sure why they showed up after you ran those commands, but your PR should only change the blog bost.

> Also, I've changed the SIG to `End-User`. I've tried shortening the title and list title. (open to changing it or shorter titles suggestions are welcome :) )

Thanks.

svrnm commented on .cspell/en-words.txt:
do not edit this file, add those words to the cspell ignore list in your document

vipinvkmenon replied:
Done

svrnm commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,601 @@
+
+### Step 2: Generate Self Signed Certificates
+
+In order to setup mTLS between the client and the server, we need a set of
+certificates. For this demo scenario, we'll be using self-signed certificate. We
+will be using the same CA for signing both clients and server in this demo.
+We'll use `openssl` to create our certificates. Please refer `openssl`
+documentation for configuration details.
```
what would I do in a non demo scenario, can you outline that or provide some perspective, can something like cert-manager be used to automate/simplify this?

svrnm commented on content/en/blog/2025/expose-otel-collector-gateway-api/plantml.txt:
is this document referenced in the blog post anywhere? Could you inline it as code block?

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,601 @@
+      as the client certificate. The Gateway should reject it during the mTLS
+      handshake because it's not signed by the trusted CA.
+
+## Caveats:
```
Done

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,601 @@
+author: >
+  [Vipin Vijaykumar](https://github.com/vipinvkmenon)
+sig: End-User SIG
+cSpell:ignore: ingressgateway otlpgrpc Vijaykumar Vipin
```
Done

vipinvkmenon commented:
Hi @svrnm @theletterf @danielgblanco,

Have made the necessary changes, and checks are also in good shape now

svrnm commented:
> Have made the necessary changes, and checks are also in good shape now

thank you. What we would need next is review and approval from the sponsors (@danielgblanco, @open-telemetry/sig-end-user-approvers ) and subject matter experts (@open-telemetry/collector-approvers and @open-telemetry/operator-approvers)

danielgblanco commented:
Thanks @vipinvkmenon! I will review this by end of this week the latest.

danielgblanco commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+- The Kubernetes Gateway API aim to make the configuration protable and
+  implementation agnostic as much as possible. Ideally this would be the case
+  onces the spec is matured and evolved. Till then, certain aspects of the
+  configuration will have minor changes between implementations. For e.g, The
```
```suggestion
  configuration will have minor changes between implementations. For example, the
```

danielgblanco commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+  implementation agnostic as much as possible. Ideally this would be the case
+  onces the spec is matured and evolved. Till then, certain aspects of the
+  configuration will have minor changes between implementations. For e.g, The
+  way `MTLS` is configured in the Gateway now.
```
```suggestion
  way mTLS is configured in the Gateway now.
```

danielgblanco commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+- When running in production, you may want to use the `infrastructure` block of
+  the `spec` to configure infrastructure-provider specific parameters. For e.g
+  `DNS`.
+- Productive setups would have end-2-end encrypted communication. For e.g, when
```
```suggestion
- Productive setups would have end-to-end encrypted communication. For example, when
```

danielgblanco commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+  the `spec` to configure infrastructure-provider specific parameters. For e.g
+  `DNS`.
+- Productive setups would have end-2-end encrypted communication. For e.g, when
+  using istio, all components running within namespaces managed by istio in a
```
```suggestion
  using Istio, all components running within namespaces managed by Istio in a
```

chalin replied:
Thanks for catching this. See: 

- #6907

danielgblanco commented:
Thanks @vipinvkmenon this is a very clear and well put hands-on example! :) I left a few comments mostly for format and consistency, and some general questions.

chalin commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+  coming into the spec. Many of these features are now in alpha/beta and will
+  soon become generally available. For e.g `GRPCRoute`. Please refer the latest
+  Kubernetes documentation of the Gateway API.
+- The Kubernetes Gateway API aim to make the configuration protable and
```
"protable" misspelling will now be reported, see:

- #6907

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+---
+title: Exposing OTel Collector in Kubernetes with Gateway API & mTLS
+linkTitle: OTel with gateway and mtls
```
That makes sense

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+export CLIENT_SUBJ="/CN=external-otel-client"
+
+# 1. Create Root CA certificate and key
+openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj -subj "${ROOT_CA_SUBJ}" -keyout rootCA.key -out rootCA.pem
```
Oopsie my bad.

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+## The Scenario
+
+1.  Deploy an OTel Collector inside Kubernetes configured with a simple OTLP
+    gRPC receiver.
```
Yes makes sense

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+openssl req -newkey rsa:4096 -nodes -keyout server.key -out server.csr -subj "${SERVER_SUBJ}" \
+  -addext "subjectAltName = DNS:${SERVER_HOSTNAME}" # Add SAN for hostname validation
+
+openssl x509 -req -in server.csr -CA rootCA.crt -CAkey rootCA.key -CAcreateserial -out server.pem -days 365 -sha256 \
```
Typo

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+- **Portability:** Standardized API definition aims for greater portability
+  across different underlying gateway/service mesh implementations.
+- **Expressiveness:** Natively supports advanced features like header
+  manipulation, traffic splitting, mTLS configuration, gRPC routing, and more.
```
My thought was that the primer might be a good recap/intro. But your point is valid. It makes more sense for the reader to understand this from the K8s documentation. I've added a link for mTLS. So, I would also support and agree to the removal of the block. In addition to the documentation links, these would keep information about these topics more up to date.

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+depending on the Gateway/service-mesh implementation you use, you may configure
+the namespace accordingly. For e.g: with istio, we need to label the namespace
+for `istio-injection:enabled` inorder for istio to work with the deployed
+workloads in the namespace
```
Yes makes sense. Noted

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+          # Use a specific, recent version tag in production
+          image: otel/opentelemetry-collector-contrib:0.119.0
+          command:
+            ['/otelcol-contrib', '--config=/conf/otel-collector-config.yaml']
```
Updated to using th non-contrib image and referred [documentation](https://opentelemetry.io/docs/collector/configuration/#location)

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
````diff
@@ -0,0 +1,602 @@
+```bash
+# Create a secret with the server certificate, key. We will also store the ca certificate used to sign clients
+# For this demo, we place it in the default namespace
+k create secret generic otel-gateway-server-cert --from-file=tls.crt=server.pem --from-file=tls.key=server.key --from-file=ca.crt=rootCA.pem
````
Updated and kept all places uniform

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+kind: Gateway
+metadata:
+  name: otel-gateway
+  namespace: default
```
Not really.. I'll mention this is just for the demo

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+spec:
+  # Corresponds to the GatewayClass installed in your cluster (e.g., istio)
+  gatewayClassName: istio
+  #infrastructure: #Use to configure infrastructure annotations for your sepcific cluster. For e.g DNS
+  #annotations:
+  #specific-anotations:value
```
Removed

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,602 @@
+      containers:
+        - name: otel-collector
+          # Use a specific, recent version tag in production
+          image: otel/opentelemetry-collector-contrib:0.119.0
```
updated to non-contrib

vipinvkmenon commented:
Hi @danielgblanco 

Thank you for your quick, thorough and insightful review. I've learnt something new...mainly the browser-client setup. 

Have incorporated the changes and use cases as well.
I have also removed the explanation for Gateway and mtls.

danielgblanco commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,613 @@
+  [tail sampling processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/tailsamplingprocessor)
+  can be configured in this central collector sitting behind the Gateway. This
+  setup can further be scaled and optimized with the
+  [load balancing exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/loadbalancingexporter).
```
In hindsight, maybe mentioning the `tailsampling` processor is enough. The `loadbalancing` exporter is definitely needed, but it's sort of an implementation detail. The important bit is that tail-sampling needs a central point to process all spans, which you covered in the previous sentence.

danielgblanco commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,613 @@
+
+    docker run --rm -v $(pwd)/certs:/etc/cert/ \
+               -v $(pwd)/otel-client-config.yaml:/etc/otelcol/config.yaml \
+               otel/opentelemetry-collector:latest
```
I believe for the client you'll need the contrib image, as the core image does not have the `hostmetrics` receiver. Might be worth giving all the scripts a final run on a clean env to make sure it all works as expected.

vipinvkmenon replied:
This is a good point. I rechecked the steps and made a few corrections and updates.

vipinvkmenon replied:
The collector (server) now uses the `otel-core` image and on the client the `otel-contrib`

vipinvkmenon commented:
Scenarios Validated:

Client Side:
<img width="1456" alt="image" src="https://github.com/user-attachments/assets/67885f82-2d79-42b2-b67f-94d082fac96e" />


Server Side:
<img width="1335" alt="image" src="https://github.com/user-attachments/assets/cac6ed8b-a1e0-4caa-9c11-7944186e9db2" />

danielgblanco commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,621 @@
+- **External Monitoring Agents:** Third-party agents or locally running
+  development instances needing to connect to a shared collector within the
+  cluster.
+- **Client-Side Monitoring** Telemetry coming in from external clients like
+  browsers and mobile applications. While mTLS may not be used in for
+  [sending metrics from the Browsers](/docs/languages/js/getting-started/browser/),
+  the collectors must ultimately be made available.
```
```suggestion
- **Client-Side Monitoring:** Telemetry coming in from external clients like
  browsers and mobile applications. While mTLS may not be used for
  [exporting telemetry from browsers](/docs/languages/js/getting-started/browser/),
  the collectors must ultimately be made available.
```

danielgblanco commented:
Thanks @vipinvkmenon! Looking good. I caught a small thing in the browser telemetry point, but after that I think we're good to go!

danielgblanco reviewed (approved):
Thanks for the corrections @vipinvkmenon, this is looking great! From my perspective as sponsor, this is ready 🚀

tiffany76 commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,621 @@
+end-to-end encryption, and aligns with Zero Trust security principles:
+
+- Ensures _only_ trusted clients (those possessing a valid certificate signed by
+  trusted CA) can connect to your exposed service.
```
```suggestion
  trusted Certificate Authority (CA)) can connect to your exposed service.
```

tiffany76 commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,621 @@
+Here are the steps we'll follow to expose an OTel Collector deployment outside
+the cluster.
+
+1.  Deploy an OTel Collector inside Kubernetes configured with a simple
```
This is a bit of a nit, but the numbering of these steps confused me because they don't line up with the "Step N" headings that follow. I would suggest changing this to an unordered list or making the ordered list match the step headings.

tiffany76 commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,621 @@
+
+![Scenario Diagram](ScenarioFlow.png)
+
+## Setup
```
I don't think we need this heading. I'd remove it and bump step 1 and step 2 up to ##-level headings. Otherwise the page nav looks a little strange:

<img width="205" alt="Screenshot 2025-05-29 at 3 13 44 PM" src="https://github.com/user-attachments/assets/b0657edc-363c-4a01-8143-b3c101c73757" />

vipinvkmenon replied:
Something like this
<img width="304" alt="image" src="https://github.com/user-attachments/assets/1da3e4eb-ada9-4a40-8a30-634a81c85641" />

tiffany76 commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,621 @@
+
+We will deploy our OTel Collector setup in the given namespace. Further on,
+depending on the Gateway/service mesh implementation you use, you may configure
+the namespace accordingly. For e.g: with Istio, we can the namespace with
```
Seems like there's a word missing from this sentence. I added "create" but that might not be correct. Also better to spell out "example" in this case.

```suggestion
the namespace accordingly. For example, with Istio, we can create the namespace with
```

tiffany76 commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,621 @@
+  configure the backend to which the route forwards the requests. In this case,
+  to the `otel-collector-server-svc`.
+
+_Note: We make use of `options` in the gateway for implementation specific
+configuration of mTLS. Currently the gateway API does not explicitly have
+`Mutual TLS`
+[mode](https://gateway-api.sigs.k8s.io/reference/spec/#tlsmodetype). Please
+refer latest documentation of Gateway API for updates_
```
```suggestion
{{% alert title="Note" %}}
We make use of `options` in the gateway for implementation-specific
configuration of mTLS. Currently, the gateway API does not explicitly have
`Mutual TLS`
[mode](https://gateway-api.sigs.k8s.io/reference/spec/#tlsmodetype). Refer 
to the latest documentation of Gateway API for updates.
{{% /alert %}}
```

tiffany76 commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,621 @@
+
+For this demo, the client (OTel Collector) is run locally with Docker.
+
+A simple configuration for scraping CPU & memory metrics and sending it to the
+server.
+
+Example `otel-client-config.yaml`:
```
```suggestion
The following example `otel-client-config.yaml` is a simple configuration for scraping CPU and memory metrics and sending them to the server:
```

tiffany76 commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,621 @@
+We are running a container of `opentelemetry-collector-contrib` by mounting the
+`otel-client-config.yaml` and the `certs` folder that contains the certificates.
+
+## Step 6: Testing the Connection
```
```suggestion
## Step 8: Testing the Connection
```

tiffany76 commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,621 @@
+  for signing the server certificates.
+- The Kubernetes Gateway API is constantly evolving with more and more features
+  coming into the spec. Many of these features are now in alpha/beta and will
+  soon become generally available. For e.g `GRPCRoute`. Please refer the latest
```
```suggestion
  soon become generally available, for example, `GRPCRoute`. Refer to the latest
```

tiffany76 commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,621 @@
+  configuration will have minor changes between implementations. For example,
+  the way mTLS is configured in the Gateway now.
+- When running in production, you may want to use the `infrastructure` block of
+  the `spec` to configure infrastructure-provider specific parameters. For e.g
+  `DNS`.
```
```suggestion
  the `spec` to configure infrastructure-provider specific parameters, for example,
  `DNS`.
```

tiffany76 commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,621 @@
+  well.
+- When working with routes and gateways in multiple namespaces, you may need
+  references of resources such as backend `services`, and other configurations
+  from other namespaces. For this please refer to Gateway
```
```suggestion
  from other namespaces. For details, refer to Gateway
```

tiffany76 commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,621 @@
+
+## Alternative Gateway Implementations
+
+While we used Istio (`gatewayClassName: istio`), the core benefits of the
```
```suggestion
While we used Istio (`gatewayClassName: istio`), the core benefit of the
```

tiffany76 reviewed (commented):
Thanks for the post, @vipinvkmenon! I've performed a copy edit review and made some suggestions. Let me know if you have any questions.

vipinvkmenon commented:
Hi @tiffany76 

I have incorporated the changes. I have moved all the Steps 1-8 under heading level `###`..As seen in screenshot above...
The idea was to club them all under setup WDYT?...Probably I could just remove `setup` and move all steps under `scenario` as well.

Also, any idea on how to fix the `BUILD * CHECK LINKS` test?

chalin commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,619 @@
+    [CLI](https://github.com/openssl/openssl/wiki/Binaries) for generating
+    certificates.
+
+{{% alert %}}
```
```suggestion
{{% alert title="API stability note" color=warning %}}
```

chalin commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,619 @@
+  For example, in a multi-cluster service mesh setup, workloads may require tail
+  sampling to be performed centrally. In this case, a central collector
+  configured with the
+  [tail sampling processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/tailsamplingprocessor)
```
Pin the URL to the most current version.

vipinvkmenon replied:
Pinning it to the latest `v0.127.0`

vipinvkmenon commented on content/en/blog/2025/expose-otel-collector-gateway-api/index.md:
```diff
@@ -0,0 +1,619 @@
+linkTitle: Collector with Gateway API and mTLS
+date: 2025-04-06
+author: >
+  [Vipin Vijaykumar](https://github.com/vipinvkmenon)
```
Sure

chalin commented:
@vipinvkmenon - you'll need to rebase and resolve conflicts with the refcache. Probably simplest to start from the version of the refcache in main, then rerunning `check:links` locally to re-populate the refcache.

> <img width="823" alt="image" src="https://github.com/user-attachments/assets/65896074-5d00-4286-92f7-a5f583faf3a8" />

vipinvkmenon commented:
Yup all updated I hope

tiffany76 reviewed (approved):
LGTM! 🎉
